### PR TITLE
Dragon camp - hide hints and fix Firebase issues

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
Commenting out sections of ITS connectivity to ensure the students can play through end-to-end

### CODE REVIEW

File: src/code/middleware/its-log.js [hunk 1]
```diff
@@ -31,19 +31,19 @@ export function initializeITSSocket(guideServer, socketPath, store) {
 
     if (event.isMatch("ITS", "HINT", "USER")) {
       // TODO: Re-enable dispatch once ITS is fixed
```
Comment (kswenson) [inline src/code/middleware/its-log.js]: These TODO messages aren't quite right at this point, but things are sufficiently fluid that it probably doesn't matter much.

File: src/code/middleware/state-save.js [hunk 2]
```diff
@@ -41,21 +41,43 @@ export default () => store => next => action => {
...
       };
     }
-    firebase.database().ref(userQueryString).update(userDataUpdate);
+    firebase.database().ref(userQueryString).update(userDataUpdate, (error, userDataUpdate) => {
+      if (error) {
+        console.logError("Error updating user state!", userDataUpdate, error);
+        return error;
+      } else {
+        return;
+      }
```
Comment (kswenson) [inline src/code/middleware/state-save.js]: I believe the `else` clause can be skipped here.

File: src/code/middleware/state-save.js [hunk 3]
```diff
@@ -41,21 +41,43 @@ export default () => store => next => action => {
...
-        remediationHistory,
+        // remediationHistory,
         stateVersion: currentStateVersion
       };
     }
-    firebase.database().ref(userQueryString).update(userDataUpdate);
+    firebase.database().ref(userQueryString).update(userDataUpdate, (error, userDataUpdate) => {
+      if (error) {
+        console.logError("Error updating user state!", userDataUpdate, error);
+        return error;
```
Comment (kswenson) [inline src/code/middleware/state-save.js]: I'm not seeing any code that's dealing with this error return value. If the point is just to log the error, then maybe the returns can be removed for simplification.

File: src/code/modules/notifications.js [hunk 4]
```diff
@@ -23,27 +23,32 @@ const initialState = {
 export default function notifications(state = initialState, action) {
   switch (action.type) {
     case GUIDE_HINT_RECEIVED:
+      // TODO: Remove when ITS is fixed
+      if (window.location.href.indexOf('enableITS') > -1) {
```
Comment (kswenson) [inline src/code/modules/notifications.js]: We have a `urlParams` module that could be used here.

File: src/code/middleware/state-save.js (hunk 3 above)
Comment (dstrawberrygirl) [reply src/code/middleware/state-save.js]: I want to include more thorough understanding of firebase connection state issues and handle failed updates more gracefully. The above code didn't even get hit when the Firebase update failed (due to undefined data being passed in the remediation section) - definitely flagging this in PT for further investigation.

**Comment by dstrawberrygirl** [comment]
Thanks @kswenson for the feedback. Will integrate these into the next PR to re-enable ITS functionality with the ability to turn it off as required. At least this build seems to be sufficiently stable that students can complete the game without blockers.

**Pull request merged**